src/reactor/mod.rs

```diff
@@ -460,14 +466,24 @@ impl Inner {
         let slot = self.timer_heap.push((at, entry.index()));
         let entry = entry.insert((Some(slot), TimeoutState::NotFired));
         debug!("added a timeout: {}", entry.index());
-        Ok((entry.index(), at))
+        return entry.index();
     }
 
     fn update_timeout(&mut self, token: usize, handle: Task) -> Option<Task> {
         debug!("updating a timeout: {}", token);
         self.timeouts[token].1.block(handle)
     }
 
+    fn reset_timeout(&mut self, token: usize, at: Instant) {
+        let pair = &mut self.timeouts[token];
+        if let Some(slot) = pair.0.take() {
+            self.timer_heap.remove(slot);
```

Comment:
Could you add a TODO here to avoid remove + push and instead just do one sift of the heap? In theory we could update it in place and then do the percolation as necessary